src/lib/ui/Switch/Switch.stories.tsx

```diff
@@ -0,0 +1,33 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import { useEffect, useState } from 'react'
+
+import Switch from './Switch.js'
+
+const meta: Meta<typeof Switch> = {
+  title: 'Library/Switch',
+  tags: ['autodocs'],
+  component: Switch,
+}
+
+type Story = StoryObj<typeof Switch>
+
+export const Content: Story = {
+  render: (props) => {
+    const [checked, setChecked] = useState(props.checked || false)
+
+    // allows the prop from the story to update the component
+    useEffect(() => {
+      setChecked(props.checked)
+    }, [props.checked])
```

Comment:
I tried this on the story and it almost works, but toggling the switch doesn't update the Storybook control to match. I wonder if there is a hook we can use from Storybook that will sync the state for us. 

Not a big deal for this story, but I'd love to understand the controls a bit better to enhance the Storybook devx overall.

Reply:
From GPT:

> Unfortunately, Storybook doesn't provide out-of-the-box two-way data binding for its controls. The controls in Storybook's UI will not automatically update when the state within a component (or a story) changes. So while you can set initial values via args and use those values inside your components, changes to internal component state will not propagate back to the controls.
> 
> However, if you truly want a mechanism where toggling the switch updates the Storybook controls, you'd likely have to make use of story actions and some additional custom setup, but this won't be a trivial task.
> 
> One simple way to provide feedback within Storybook without changing the controls would be to use an action to log switch state changes. Actions can be used to display data in the "Actions" panel in Storybook:

I also wanted the storybook control to update but it might be tough. lemme look more into it.

Reply:
Ok, this is very much a nice to have. Mostly something I've been curious about but haven't had the opportunity to dig further myself.

Reply:
Oh baby!

Reply:
https://sandroroth.com/blog/storybook-controlled-components

Reply:
https://github.com/seamapi/react/pull/377/commits/ea36ce2ba9d59df08e9b2616ac49878a31cfd1b6#diff-74b781a75c672defb3e830dd400bb08c08c6f4b807c10957db4593d8726a8ff6R1-R30